test/index.js

```diff
@@ -93,6 +93,29 @@ describe( 'I18n', function() {
 					count: 2
 				} ) );
 			} );
+
+			describe( 'adding new translations', function() {
+				it( 'should find a new translation after it has been added', function() {
+					i18n.addTranslations( {
+						'test3': [
+							null,
+							'translation3'
+						]
+					} );
+
+					assert.equal( 'translation3', translate( 'test3' ) );
+				} );
+				it( 'should return the new translation if it has been overwritten', function() {
+					i18n.addTranslations( {
+						'test1': [
```

Comment:
It would be better to use a fresh variable here.   If this test ran first, `should find a simple translation` would fail.